ballerina/types.bal

```diff
@@ -8,66 +12,84 @@ public type NameConfig record {|
+    # This field is only considered if the expected type is a subset of `anydata[][]`
+    boolean outputWithHeaders = false;
 };
 
-public type ParseOption record {|
-  *Options;
-  string:Char delimiter = ",";
-  string encoding = "UTF-8";
-  string locale = "en_US";
-//  TODO: Add " for Strings"
-  string:Char textEnclosure = "\"";
-  string:Char escapeChar = "\\";
-  LineTerminator|LineTerminator[] lineTerminator = [CR, LF, CRLF];
-  NilValue? nilValue = ();
-  // string commentStartingSequence = "#";
-  string:Char comment = "#";
-  false|int:Unsigned32 header = 0;
-|};
-
-public type parseToRecordOption record {|
-  *ParseOption;
-
-  // if header = false and this value is null, Then compiler time error.
-  string[]? customHeaders = ();
-   boolean enableConstraintValidation = true;
-|};
-
-public type ListAsListOption record {|
+# Represents the options for parsing data.
+public type ParseOptions record {|
     *Options;
-    boolean stringConversion = true;
+    # The delimiter character used for separating fields in the data.
+    string:Char delimiter = ",";
+    # The character encoding of the data.
+    string encoding = "UTF-8";
+    # The locale used for parsing.
+    string locale = "en_US";
+    # The character used to enclose text fields.
+    string:Char textEnclosure = "\"";
+    # The character used for escaping.
+    string:Char escapeChar = "\\";
+    # The line terminator(s) used in the data.
+    LineTerminator|LineTerminator[] lineTerminator = [LF, CRLF];
+    # The value to represent nil.
+    NilValue? nilValue = ();
+    # The character used to indicate comments in the data.
+    string:Char comment = "#";
+    # Specifies whether the header is present and, if so, the number of header lines.
+    false|int:Unsigned32 header = 0;
+    # Custom headers for the data, if headers are absent.
+    string[]? customHeadersIfHeaderAbsent = ();
 |};
 
-public type RecordAsRecordOption record {|
+# Represents options for treating a list as a record.
+public type ParseListOptions record {|
     *Options;
-   boolean enableConstraintValidation = true;
+    # If `0`, all the source data will treat as data rows.
+    # Otherwise specify the header rows(Starts from 1) in the source data.
+    int:Unsigned32 headerRows = 0;
+    # Specify the header names of the source data.
+    # This field will overwrite the header values in the header rows.
+    # This will be mandatory if the header row parameter is larger than one.
+    string[]? customHeaders = ();
 |};
 
-public type ListAsRecordOption record {|
+# Represents options for treating a list as a record.
+public type TransformOptions record {|
     *Options;
-    boolean enableConstraintValidation = true;
-    boolean stringConversion = true;
+    # Specify the order of the headers in the source data.
+    # If the expected type is a subset of `record{}[]` this parameter will be ignored.
+    string[]? headersOrder = ();
```

Comment:
Please check for naming consistency. E.g., headerRows in L67 vs headersOrder here.

Reply:
I will update  `headersOrder` as `headerRows`